packages/recommend-core/src/personalization/getPersonalizationFilters.ts

```diff
@@ -0,0 +1,54 @@
+import { getAffinities } from './getAffinities';
+import { getStrategy } from './getStrategy';
+
+type GetPersonalizationFilters = {
+  userToken: string;
+  region: string;
+  apiKey: string;
+  appId: string;
+};
+
+export const getPersonalizationFilters = async ({
+  userToken,
+  region,
+  apiKey,
+  appId,
+}: GetPersonalizationFilters) => {
+  if (!region || !apiKey || !appId) {
+    throw new Error(
+      `[Algolia Recommend] parameters 'region', 'apiKey' and 'appId' are required to enable personalization.`
+    );
+  }
+  if (!userToken) {
+    // eslint-disable-next-line no-console
+    console.warn(
+      `[Algolia Recommend] parameter 'userToken' is required to enable personalization.`
```

Comment:
```suggestion
      `[Algolia Recommend] Personalization couldn't be enabled because `userToken` is missing. Falling back to non-personalized recommendations.`
```